src/event.rs

```diff
@@ -106,6 +106,49 @@ pub enum Event {
     #[cfg(feature = "xv")]
     /// The event is issued from the `XVideo` extension.
     Xv(xv::Event),
+
+    /// The event was not recognized, it was likely issued from a disabled extension.
+    Unresolved(UnresolvedEvent),
+}
+
+/// an event was not recognized as part of the core protocol or any enabled extension
+pub struct UnresolvedEvent {
+    raw: *mut xcb_generic_event_t,
+}
+
+impl BaseEvent for UnresolvedEvent {
+    const EXTENSION: Option<Extension> = None;
+    const NUMBER: u32 = u32::MAX;
+
+    unsafe fn from_raw(raw: *mut xcb_generic_event_t) -> Self {
+        UnresolvedEvent { raw }
+    }
+
+    unsafe fn into_raw(self) -> *mut xcb_generic_event_t {
+        let raw = self.raw;
+        std::mem::forget(self);
+        raw
+    }
+
+    fn as_raw(&self) -> *mut xcb_generic_event_t {
+        self.raw
+    }
+
+    fn as_slice(&self) -> &[u8] {
+        &[]
```

Comment:
This is incorrect.
The purpose of `as_slice` is to provide the raw event data in the form of `[u8]`.
I would agree that the name could be better chosen (I will probably change this before the final release)

Reply:
I understood the purpose of the method but i thought that since we don't know what the concrete type is we can't know how much we can safely read from the pointer so opted for the safest variant. Is there a way to know without resolving the concrete event type?

Reply:
You can see [here](https://rust-x-bindings.github.io/rust-xcb/branches/v1.0-dev/src/xcb/home/runner/work/rust-xcb/rust-xcb/target/debug/build/xcb-8df0b2892b11bc21/out/xproto.rs.html#602) for an example. (for the length, `BaseEvent` by definition are 32 bytes long)

Reply:
Ah ok, should I implement the Wired trait for it then as well? Also now I'm thinking, is there a posibility that the unresolved event was not a base event but a ge event, if it likely belonged to a disabled extension where ge events are used?

Reply:
`Wired` is not strictly necessary. It is used to serialize/deserialize structs and values. In the case of `x::SendEvent`, the event is serialized using `as_slice` (see [here](https://rust-x-bindings.github.io/rust-xcb/branches/v1.0-dev/src/xcb/home/runner/work/rust-xcb/rust-xcb/target/debug/build/xcb-8df0b2892b11bc21/out/xproto.rs.html#14194))

Regarding GE events, they all have `response_type == XCB_GE_GENERIC`. In your current code, unresolved GE events would panic. (see [here](https://github.com/pr2502/rust-xcb/blob/ae0f2ed385fbd0b9908ac241c20fc5286f1ed934/src/event.rs#L174))

Reply:
> we can't know how much we can safely read from the pointer so opted for the safest variant

This is a good remark. The only way for the user to construct `UnknownEvent` is `BaseEvent::from_raw`, which is unsafe.
I will add in the safety doc section that the data must be 32 bytes long. This is implied by being a valid event though.

Reply:
thank you a lot for the explanation and links! i've implemented it now and included comments to hopefully make it clear why is it ok to read 32 bytes there